File: php-sdk/src/Fyipe/FyipeTimelineManager.php
<?php

/**
 * @author bunday
 */

namespace Fyipe;


class FyipeTimelineManager
{
    private $options=[];
    private $timeLineStack = [];

    /**
     * FyipeTimelineManager constructor.
     * @param array $options
     */
    public function __construct($options){
        $this->options = $options;
    }
    private function addItemToTimeline($item) {
        // get the size of the stack
        if (isset($this->options['maxTimeline']) && (sizeof($this->timeLineStack) === $this->options['maxTimeline'])) {
            return; // It discards new timline update once maximum is reached
        }
        // add time to it
        $item->timestamp = time();
        // add a new item to the stack
        array_push($this->timeLineStack, $item);
        return true;
    }
    public function addToTimeline($item) {
        $this->addItemToTimeline($item);
    }
     // return the timeline
     public function getTimeline() {
        return $this->timeLineStack;
    }
    // clear the timeline
    public function clearTimeline() {
        $this->timeLineStack = [];
    }
}

File: java-sdk/src/main/java/io/hackerbay/fyipe/FyipeTimelineManager.java
package io.hackerbay.fyipe;

import io.hackerbay.fyipe.model.Timeline;
import io.hackerbay.fyipe.model.TrackerOption;

import java.sql.Timestamp;
import java.util.ArrayList;

public class FyipeTimelineManager {
    private TrackerOption options;
    private ArrayList<Timeline> timelineStack = new ArrayList<Timeline>();
    private Timestamp timestamp = new Timestamp(System.currentTimeMillis());

    public FyipeTimelineManager(TrackerOption options) {
        this.options = options;
    }
    public void addItemToTimeline(Timeline item) {
        // get the size of the stack
        if(this.timelineStack.size() == this.options.getMaxTimeline()) {
            return; // It discards new timeline update once maximum is reached
        }
        // add time to the timeline item
        item.setTime(timestamp.getTime());

        // add a new item to the stack
        this.timelineStack.add(item);
        return;
    }
    public ArrayList<Timeline> getTimeline() {
        return this.timelineStack;
    }
    public void clearTimeline() {
        this.timelineStack.clear();
    }

}
